scipy/stats/_continuous_distns.py

```diff
@@ -6121,8 +6122,75 @@ def _entropy(self, c):
     def fit(self, data, *args, **kwds):
         parameters = _check_fit_input_parameters(self, data, args, kwds)
         data, fshape, floc, fscale = parameters
-        if floc is None:
-            return super().fit(data, **kwds)
+
+        if floc is None and fscale is None:
+            # Optimization over a system of equations is possible when the
+            # location of the data is not set. The optimization requires that
+            # the scale be free, but the shape may be fixed.
+
+            def get_shape(scale, location):
+                # Return the fixed shape or numerically optimal shape based on
+                # the input parameters and data.
+                if fshape is None:
+                    return (data.shape[0] /
+                            np.sum(np.log((data - location) / scale)))
+                else:
+                    return fshape
+
+            def get_location(scale):
+                # Return the numerically optimal location.
+                return np.min(data) - scale
+
+            def dLdScale(shape, scale):
+                # The partial derivative of the log-likelihood function w.r.t.
+                # the scale.
+                return data.shape[0] * shape / scale
+
+            def dLdLocation(shape, location):
+                # The partial derivative of the log-likelihood function w.r.t.
+                # the location.
+                return (shape + 1) * np.sum(1 / (data - location))
+
+            def fun_to_solve(scale):
+                # optimize over the scale, setting the solving for the root
+                location = get_location(scale)
+                shape = get_shape(scale, location)
+                return (dLdLocation(shape, location) - dLdScale(shape, scale)) ** 2
```

Comment:
```suggestion
                return dLdLocation(shape, location) - dLdScale(shape, scale)
```
leftover from a bit of experimentation.